src/FEValues/FunctionValues.jl

```diff
@@ -0,0 +1,78 @@
+# Helpers to get the correct types for FunctionValues for the given function and, if needed, geometric interpolations. 
+struct SInterpolationDims{rdim,sdim} end
+struct VInterpolationDims{rdim,sdim,vdim} end
+function InterpolationDims(::ScalarInterpolation, ip_geo::VectorizedInterpolation{sdim}) where sdim
+    return SInterpolationDims{getdim(ip_geo),sdim}()
+end
+function InterpolationDims(::VectorInterpolation{vdim}, ip_geo::VectorizedInterpolation{sdim}) where {vdim,sdim}
+    return VInterpolationDims{getdim(ip_geo),sdim,vdim}()
+end
+
+typeof_N(::Type{T}, ::SInterpolationDims) where T = T
+typeof_N(::Type{T}, ::VInterpolationDims{<:Any,dim,dim}) where {T,dim} = Vec{dim,T}
+typeof_N(::Type{T}, ::VInterpolationDims{<:Any,<:Any,vdim}) where {T,vdim} = SVector{vdim,T} # Why not ::Vec here?
+
+typeof_dNdx(::Type{T}, ::SInterpolationDims{dim,dim}) where {T,dim} = Vec{dim,T}
+typeof_dNdx(::Type{T}, ::SInterpolationDims{<:Any,sdim}) where {T,sdim} = SVector{sdim,T} # Why not ::Vec here?
+typeof_dNdx(::Type{T}, ::VInterpolationDims{dim,dim,dim}) where {T,dim} = Tensor{2,dim,T}
+typeof_dNdx(::Type{T}, ::VInterpolationDims{<:Any,sdim,vdim}) where {T,sdim,vdim} = SMatrix{vdim,sdim,T} # If vdim=sdim!=rdim Tensor would be possible...
+
+typeof_dNdξ(::Type{T}, ::SInterpolationDims{dim,dim}) where {T,dim} = Vec{dim,T}
+typeof_dNdξ(::Type{T}, ::SInterpolationDims{rdim}) where {T,rdim} = SVector{rdim,T} # Why not ::Vec here?
+typeof_dNdξ(::Type{T}, ::VInterpolationDims{dim,dim,dim}) where {T,dim} = Tensor{2,dim,T}
+typeof_dNdξ(::Type{T}, ::VInterpolationDims{rdim,<:Any,vdim}) where {T,rdim,vdim} = SMatrix{vdim,rdim,T} # If vdim=rdim!=sdim Tensor would be possible...
```

Comment:
This should probably be `Tensor`s in a few more places as commented in the future. For this PR; keeping as no change wrt. master (contractions give dimension mismatch otherwise).

Reply:
I decided to consistently use static arrays for embedded stuff and for the case that vdim != sdim , because Tensors.jl does does not allow tensors with weird shapes (e.g. 2x3 tensor) for now (although I know you work on it :)). I think returning some Tensors.jl stuff and some StaticArrays stuff causes more confusion than having the two consistens interfaces which are in line with their applications, i.e. Tensors.jl for mechanics stuff, StaticArrays for the remaining cases.